gp2gp-translator/src/test/java/uk.nhs.adaptors.pss.translator/Generator/BundleGeneratorTest.java

```diff
@@ -0,0 +1,44 @@
+package uk.nhs.adaptors.pss.translator.Generator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import uk.nhs.adaptors.pss.translator.generator.BundleGenerator;
+import uk.nhs.adaptors.pss.translator.service.FhirIDGeneratorService;
+import uk.nhs.adaptors.pss.translator.utils.ResourceTestFileUtils;
+
+@SpringBootTest
+@ExtendWith({SpringExtension.class})
+@DirtiesContext
+public class BundleGeneratorTest {
+
+    private static final FhirIDGeneratorService fhirIDGeneratorService = new FhirIDGeneratorService();
+    private static final BundleGenerator bundleGenerator = new BundleGenerator(fhirIDGeneratorService);
+    private static final String BUNDLE_RESOURCE = "resources/generator/bundleResource.json";
+
+    private static Stream<Arguments> resourceFileParams() {
+        return Stream.of(
+            Arguments.of(BUNDLE_RESOURCE)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("resourceFileParams")
+    public void When_GeneratingBundleResource_Expect_BundleResourceJson(String outputBundle)
+        throws IOException {
+
+        CharSequence expectedOutputJson = ResourceTestFileUtils.getFileContent(outputBundle);
+        assertThat(bundleGenerator.generateBundle()).isEqualTo(expectedOutputJson);
```

Comment:
I don't think this test will pass - id is randomly generated, but hardcoded in bundleResource.json. You have to mock the fhirIDGeneratorService.generateUuid() response, so it returns the value expected in bundleResource.json